UnityProject/Assets/Scripts/Objects/Directionals/DirectionalPassable.cs

```diff
@@ -1,105 +1,204 @@
+		{
+			if (!Passable) return false;
+			if (IsEnterableOnAll) return true;
+			if (context == gameObject || context == null) return true;
 
-	public void DenyPassableOnAllSides()
-	{
-		passableOnAll = false;
-		Passable = false;
-	}
+			return IsPassableAtSide(GetSideFromVector(enteringFrom), enterableSides);
+		}
 
-	public void AllowPassableAtSetSides()
-	{
-		passableOnAll = false;
-		Passable = true;
-	}
+		public override bool IsPassableTo(Vector3Int leavingTo, bool isServer, GameObject context = null)
+		{
+			if (!Passable) return false;
+			if (IsLeavableOnAll) return true;
+			if (context == gameObject || context == null) return true;
 
-	bool CheckPassable(Vector3Int sideVector)
-	{
-		// If passable on all, can be passable on any side.
-		if (PassableOnAll) return true;
-		// If not passable, cannot be passed on any side.
-		if (!Passable) return false;
-
-		Vector2Int sideToCross = (sideVector - LocalPosition).To2Int();
-		bool result = IsPassableAtSide(sideToCross);
-		return result;
-	}
+			return IsPassableAtSide(GetSideFromVector(leavingTo), leavableSides);
+		}
 
-	bool IsPassableAtSide(Vector2Int sideToCross)
-	{
-		// TODO Consider using some data structure
-		switch (Directional.CurrentDirection.AsEnum())
+		public override bool IsAtmosPassable(Vector3Int enteringFrom, bool isServer)
 		{
-			case OrientationEnum.Up:
-				if (sideToCross == Vector2Int.up) return PassableDown;
-				else if (sideToCross == Vector2Int.down) return PassableUp;
-				else if (sideToCross == Vector2Int.left) return PassableRight;
-				else if (sideToCross == Vector2Int.right) return PassableLeft;
-				break;
-			case OrientationEnum.Down:
-				if (sideToCross == Vector2Int.up) return PassableUp;
-				else if (sideToCross == Vector2Int.down) return PassableDown;
-				else if (sideToCross == Vector2Int.left) return PassableLeft;
-				else if (sideToCross == Vector2Int.right) return PassableRight;
-				break;
-			case OrientationEnum.Left:
-				if (sideToCross == Vector2Int.up) return PassableLeft;
-				else if (sideToCross == Vector2Int.down) return PassableRight;
-				else if (sideToCross == Vector2Int.left) return PassableDown;
-				else if (sideToCross == Vector2Int.right) return PassableUp;
-				break;
-			case OrientationEnum.Right:
-				if (sideToCross == Vector2Int.up) return PassableRight;
-				else if (sideToCross == Vector2Int.down) return PassableLeft;
-				else if (sideToCross == Vector2Int.left) return PassableUp;
-				else if (sideToCross == Vector2Int.right) return PassableDown;
-				break;
+			if (!AtmosPassable) return false;
+			if (IsAtmosPassableOnAll) return true;
+
+			return IsPassableAtSide(GetSideFromVector(enteringFrom), atmosphericPassableSides);
+		}
+
+		#endregion RegisterObject Overrides
+
+		/// <summary>
+		/// Sets all sides for this object as passable for the given pass type.
+		/// </summary>
+		public void AllowPassableOnAllSides(PassType passType)
+		{
+			switch (passType)
```

Comment:
Codacy found an issue: [Add a 'default' clause to this 'switch' statement.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6377637)